core_engine/find_all_url_data.py

```diff
@@ -0,0 +1,169 @@
+import sys
+from bs4 import BeautifulSoup
+import requests
+from text_finder import read_url
+import time
+import sys
+
+sys.setrecursionlimit(100000)
+
+
+def get_all_data_from_url(url, links=[], count=0):
+    try:
+        """Handle multiple repeating of exception"""
+        print("Link...", end=' ')
+        if count > 20:
+            return 0
+        """Reading html page and extract tags with texts"""
+        response = requests.get(url).text
+        soup = BeautifulSoup(response, 'html.parser')
+        th_tags = soup.find_all('th')
+
+        """Check whether list of texts is empty and handle error 503"""
+        if not th_tags:
+            print('\033[1;31;40m Empty list. Wait a bit... \033[0m', end=' ')
+            time.sleep(2)
+            count += 1
+            return get_all_data_from_url(url, links, count)
+
+        """Search for text and add it into a file"""
+        with open("training_data.txt", 'ab') as file:
+            print('Read text', end=' ')
+            file.write(read_url(soup).encode('cp1251', errors='ignore'))
```

Comment:
utf-8 is preferred

Reply:
I had plenty of problems with utf8 and Ukrainian language. Since this program is all about collecting ukr.lang. data and cp1251 works ok, I decided to use it. If there is more important reason to use utf8, I’ll try to handle this.

Reply:
if decode text in cp1251 we might have issues in future when data will be returned to users